addons/stock/models/stock_move_line.py

```diff
@@ -497,7 +497,8 @@ def _free_reservation(self, product_id, location_id, quantity, lot_id=None, pack
                 ('product_qty', '>', 0.0),
                 ('id', 'not in', ml_to_ignore.ids),
             ]
-            oudated_candidates = self.env['stock.move.line'].search(oudated_move_lines_domain)
+            current_picking_first = lambda cand: cand.picking_id != self.move_id.picking_id
+            oudated_candidates = self.env['stock.move.line'].search(oudated_move_lines_domain).sorted(current_picking_first)
```

Comment:
we could use this commit to finally fix the typo oudated_candidates -> outdated_candidates